pandas/core/series.py

```diff
@@ -2638,15 +2639,19 @@ def combine(self, other, func, fill_value=None):
         elif is_extension_array_dtype(self.values):
             # The function can return something of any type, so check
             # if the type is compatible with the calling EA.
+            non_na_values = [v for v in new_values if notna(v)]
+            if non_na_values:
```

Comment:
this does not belong here, again why doesn't this work in infer_dtype_from?

Reply:
`infer_dtype_from` internally casts to Numpy dtype, so it will return float if it sees any nans in the array.